frontend/components/color_encoding_to_sound/BasePage.js

```diff
@@ -2,63 +2,123 @@ import React from "react";
             this.setState({
                 result: responseDict,
-                listOfColors: this.state.listOfColors.concat(tempColor),
+                instrumentGenerated: true
             });
-            console.log(this.state.listOfColors);
+            alert("Instrument has been updated!");
         };
         fetchPost('/api/color/', requestBody, responseCallbackFunc);
     };
 
     render() {
         const colorDisplay = this.state.listOfColors.map( (color, i) =>
             <div key={i}>
-                <p>Color {i+1}</p>
-                <p> R: {color.r} G: {color.g} B: {color.b}</p>
+                    <PaletteColor id={i} color={color} selected={i===Number(this.state.selected)} handlePaletteClick={this.handlePaletteClick}/>
             </div>
         );
         return (<div>
-            <SketchPicker
-                color={this.state.color}
-                onChangeComplete={this.handleChangeComplete}
-            />
-            <button onClick={this.handleSubmit}>
-                Submit
-            </button>
+            <h1> Color Encoding to Sound </h1>
+            <p> Use the color picker below to choose a color, and hit the <b>submit</b> button
+            to add up to seven colors to your palette. When you're ready to  hear them together,
+            click the <b>Generate Instrument</b> button to hear them together!</p>
+            <div className="row">
+                <SketchPicker
+                    color={this.state.colorPickerColor}
+                    onChangeComplete={this.handleChangeComplete}
+                    disableAlpha
+                    className="col-sm"
+                />
 
-            {/*<audio controls controlsList={"nodownload"}>*/}
-            {/*        <source src={`data:audio/wav;base64,${this.state.result}`} type={"audio/wav"}/>*/}
-            {/*</audio>*/}
-            {/*<audio controls="controls"*/}
-            {/*       src={`data:audio/wav;base64, ${this.state.result}`}*/}
-            {/*       controlsList="nodownload"/>*/}
+                {/*<audio controls controlsList={"nodownload"}>*/}
+                {/*        <source src={`data:audio/wav;base64,${this.state.result}`} type={"audio/wav"}/>*/}
+                {/*</audio>*/}
+                {/*<audio controls="controls"*/}
+                {/*       src={`data:audio/wav;base64, ${this.state.result}`}*/}
+                {/*       controlsList="nodownload"/>*/}
 
-            <SliderInstrument
-                samples={this.state.result}
-            />
+                <div className="col-sm-2">
+                    { colorDisplay }
+                    <button onClick={this.handleSubmit}>
+                        {this.state.instrumentGenerated ? "Update Instrument" : "Generate Instrument"}
+                    </button>
+                </div>
 
-            { colorDisplay }
+                <div className="col-sm">
+                    {this.state.result.length !== 0 && <SliderInstrument
```

Comment:
Nitpick: I find this kind of thing easier to read like this...
```
{this.state.result.length !== 0 && 
   <SliderInstrument samples={this.state.result} />
}
```
... such that the line breaks are aligned with the syntax of the expression.